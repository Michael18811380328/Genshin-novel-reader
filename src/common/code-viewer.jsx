import React from "react";
import PropTypes from "prop-types";
import SyntaxHighlighter from 'react-syntax-highlighter';
import { docco } from 'react-syntax-highlighter/dist/esm/styles/hljs';
import { dark } from 'react-syntax-highlighter/dist/esm/styles/prism';
import { getSuffix } from '../utils';

function CodeViewer (props) {
  let lan = getSuffix(props.currentFile.name);
  // TODO js 特殊处理，其他的后缀暂时不需要处理
  if (lan === 'js') {
    lan = 'javascript';
  }
  return (
    <div>
      <SyntaxHighlighter
        language={lan}
        // todo user can change theme color
        style={lan === 'javascript' ? dark : docco}
        showLineNumbers={true}
        showInlineLineNumbers={true}
        wrapLongLines={true}
      >
        {props.context}
      </SyntaxHighlighter>
    </div>
  );
}

CodeViewer.propTypes = {
  context: PropTypes.string.isRequired,
  currentFile: PropTypes.object.isRequired,
};

export default CodeViewer;
