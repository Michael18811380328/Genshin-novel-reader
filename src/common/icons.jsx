
// 付费图标
// <span class="icon icon-star-full"></span>
// 金色星星
// 点击后，弹出一个对话框，显示购买信息

// 点击某个功能，先弹出一个 popover 然后点击内部的购买信息按钮
// 再次弹出一个对话框，显示详细的购买信息


// 加载图标
{/* <span class="icon icon-spinner"></span> */}

// 搜索图标
{/* <span class="icon icon-search"></span> */}


// 分享图标
{/* <span class="icon icon-share2"></span> */}
// 点击后弹出对话框，显示一个当前页面的二维码
