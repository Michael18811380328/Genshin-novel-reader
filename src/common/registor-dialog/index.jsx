import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from "axios";
import cookie from 'react-cookies';
import { Modal, ModalHeader, ModalBody, Button, Input, Label, Form, FormGroup } from 'reactstrap';
import setting from "../../setting.json";
import toaster from '../toast';

export default class RegistorDialog extends Component {

  static propTypes = {
    toggle: PropTypes.func.isRequired
  };

  constructor (props) {
    super(props);
    this.emailRef = React.createRef();
    this.passwordRef = React.createRef();
  }

  onClick = () => {
    const email = this.emailRef.current.value.trim();
    const password = this.passwordRef.current.value;
    const options = { email, password };
    // TODO loading
    axios.post(`${setting.server}/registor`, options).then(res => {
      if (res.data.token) {
        toaster.success(`用户 ${email} 注册成功`);
        // TODO: set user permission and env
        this.saveToken(res.data.token);
        this.props.toggle();
      } else {
        toaster.danger('登录失败，请检查你的用户名和密码是否正确');
      }
    }).catch((err) => {
      if (err.response && err.response.status === 400) {
        toaster.danger('登录失败，请检查你的用户名和密码是否正确');
      } else {
        toaster.danger(err);
      }
    });
  };

  saveToken = (token) => {
    cookie.save('novelToken', token, { path: '/' });
    setTimeout(() => {
      // TODO react-router 实现跳转页面，现在不好
      // 本地开发时 "homepage": "/novel-demo/", 应该设置根目录，打包后应该改成这个目录
      window.location.href = window.location.href + 'reader'; // redirect to novel main page
    }, 1000);
  };

  render () {
    return (
      <Modal isOpen={true} toggle={this.props.toggle} className="login-dialog">
        <ModalHeader toggle={this.props.toggle}>注册</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup>
              <Label>邮箱</Label>
              <Input type="text" innerRef={this.emailRef} autoFocus />
            </FormGroup>
            <FormGroup>
              <Label>密码</Label>
              <Input type="password" innerRef={this.passwordRef}/>
            </FormGroup>
          </Form>
          <Button color="success" onClick={this.onClick}>注册</Button>
        </ModalBody>
      </Modal>
    );
  }
}
