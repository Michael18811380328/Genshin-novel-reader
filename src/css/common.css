/* common style */
.text-truncate{
  overflow:hidden;
  text-overflow:ellipsis;
  white-space:nowrap;
}

.error {
  color: red;
}

.center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.d-flex {
  display: flex;
}

.d-block {
  display: block;
}

/* 这个放在 setting 中 */
.setting-divide-line {
  margin: 14px 16px;
  border-bottom: 1px solid #e2e2e2;
}
