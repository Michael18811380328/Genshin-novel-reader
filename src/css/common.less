@import "../css/variables.less";

.text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.error {
  color: @error-color;
}

.center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.d-flex {
  display: flex;
}

.d-block {
  display: block;
}

.icon {
  color: @icon-color;
}

.icon:hover {
  color: @icon-hover-color;
}
