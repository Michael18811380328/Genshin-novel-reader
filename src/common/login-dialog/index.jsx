import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from "axios";
import cookie from 'react-cookies';
import { Modal, ModalHeader, ModalBody, Button, Input, Label, Form, FormGroup } from 'reactstrap';
import setting from "../../setting.json";
import toaster from '../toast';
import { LoadingIcon } from '../icons';

class LoginDialog extends Component {

  constructor (props) {
    super(props);
    this.emailRef = React.createRef();
    this.passwordRef = React.createRef();
    this.state = {
      isLoading: false,
    };
  }

  onLogin = () => {
    const email = this.emailRef.current.value.trim();
    const password = this.passwordRef.current.value;
    const options = { email, password };
    this.setState({ isLoading: true });
    axios.post(`${setting.server}/api/login`, options).then(res => {
      if (res.data.token) {
        toaster.success(`用户 ${email} 登录成功, 1S后跳转到阅读器`);
        // TODO: set user permission and env
        this.saveToken(res.data.token);
        this.props.toggle();
        cookie.save('username', email);
      } else {
        toaster.danger('登录失败，请检查你的用户名和密码是否正确');
      }
      this.setState({ isLoading: false });
    }).catch((err) => {
      if (err.response && err.response.status === 400) {
        toaster.danger('登录失败，请检查你的用户名和密码是否正确');
      } else {
        toaster.danger(String(err));
      }
      this.setState({ isLoading: false });
    });
  };

  saveToken = (token) => {
    cookie.save('novelToken', token, { path: '/' });
    setTimeout(() => {
      window.location.href = window.location.href + 'reader'; // redirect to novel main page
    }, 1000);
  };

  render () {
    return (
      <Modal isOpen={true} toggle={this.props.toggle} className="login-dialog">
        <ModalHeader toggle={this.props.toggle}>登录</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup>
              <Label>邮箱</Label>
              <Input type="text" innerRef={this.emailRef} autoFocus />
            </FormGroup>
            <FormGroup>
              <Label>密码</Label>
              <Input type="password" innerRef={this.passwordRef}/>
            </FormGroup>
          </Form>
          {this.state.isLoading
            ? <Button color="success" disabled><LoadingIcon/></Button>
            : <Button color="success" onClick={this.onLogin}>登录</Button>
          }
        </ModalBody>
      </Modal>
    );
  }
}

LoginDialog.propTypes = {
  toggle: PropTypes.func.isRequired
};

export default LoginDialog;
