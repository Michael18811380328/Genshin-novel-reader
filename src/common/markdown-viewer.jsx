import React from "react";
import PropTypes from "prop-types";
import ReactMarkdown from "react-markdown";
import remarkGfm from "remark-gfm"; // support for strikethrough, tables, tasklists and URLs directly
import remarkMath from "remark-math";
import rehypeKatex from "rehype-katex";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter"; // syntax highlighting
import { dark } from "react-syntax-highlighter/dist/esm/styles/prism";
import "katex/dist/katex.min.css";

// other plugins refrence https://github.com/remarkjs/remark/blob/main/doc/plugins.md#list-of-plugins

function MarkdownViewer(props) {
  return (
    <ReactMarkdown
      children={props.context}
      remarkPlugins={[remarkMath, remarkGfm]}
      rehypePlugins={[rehypeKatex]}
      components={{
        code({ node, inline, className, children, ...props }) {
          const match = /language-(\w+)/.exec(className || "");
          return !inline && match ? (
            <SyntaxHighlighter
              {...props}
              children={String(children).replace(/\n$/, "")}
              style={dark}
              language={match[1]}
              PreTag="div"
            />
          ) : (
            <code {...props} className={className}>
              {children}
            </code>
          );
        },
      }}
    />
  );
}

MarkdownViewer.propTypes = {
  context: PropTypes.string.isRequired,
};

export default MarkdownViewer;
