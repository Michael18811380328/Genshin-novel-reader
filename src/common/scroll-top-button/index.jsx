import React, { Component } from "react";
import PropTypes from "prop-types";
import intl from "react-intl-universal";
import { UncontrolledTooltip } from "reactstrap";
import "./index.css";

class ScrollTopIcon extends Component {
  render() {
    const id = "novel-scroll-top-icon";
    return (
      <>
        <div
          className="scroll-top-icon"
          onClick={this.props.onClick}
          style={this.props.style}
          id={id}
        ></div>
        <UncontrolledTooltip
          placement="right"
          target={id}
          fade={false}
          delay={{ show: 0, hide: 0 }}
        >
          {intl.get('Back to top')}
        </UncontrolledTooltip>
      </>
    );
  }
}

ScrollTopIcon.propTypes = {
  onClick: PropTypes.func.isRequired,
  style: PropTypes.object.isRequired,
};

export default ScrollTopIcon;
