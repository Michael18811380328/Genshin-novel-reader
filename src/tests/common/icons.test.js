import renderer from 'react-test-renderer';
import { StarIcon, LoadingIcon, SearchIcon, ShareIcon, RightIcon, LeftIcon, ListIcon, TreeIcon, BackIcon, DeleteIcon } from '../../common/icons';

it('render icons', () => {
  const component = renderer.create(
    <div>
      <StarIcon/>
      <LoadingIcon/>
      <SearchIcon/>
      <ShareIcon/>
      <RightIcon/>
      <LeftIcon/>
      <ListIcon/>
      <TreeIcon/>
      <BackIcon/>
      <DeleteIcon/>
    </div>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
