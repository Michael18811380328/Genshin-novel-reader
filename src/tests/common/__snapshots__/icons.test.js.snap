// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render icons 1`] = `
<div>
  <span
    className="icon icon-star-full"
  />
  <span
    className="icon icon-spinner"
  />
  <span
    className="icon icon-search"
  />
  <span
    className="icon icon-share2"
  />
  <span
    className="icon icon-arrow-right2"
  />
  <span
    className="icon icon-arrow-left2"
  />
  <span
    className="icon icon-list2"
  />
  <span
    className="icon icon-tree"
  />
  <span
    className="icon icon-arrow-left2"
  />
  <span
    className="icon icon-cancel-circle"
  />
</div>
`;
